usagov/vote-gov-drupal
# VOTE-1974: Use partial splits and complete split settings

## Jira ticket

[VOTE-1974](https://bixal-projects.atlassian.net/browse/VOTE-1974)

## Description

Added additional configuration for config split to better facilitate exporting and managing configuration files without any unnecessary file removals or review.

## Deployment and testing

### Pre-deploy

1. None

### Post-deploy

1. For testing locally confirm that `$config['config_split.config_split.local']['status'] = TRUE;` is set within your web/sites/settings.local.php file.

### QA/Test

1. From the terminal run `lando drush cim` to import the new configurations
2. Confirm that the correct configurations are imported. _If you are already imported the selected environment's configs they you may not see any changes at all._
3. Uninstall DBlog `lando drush pmu dblog'
4. Export the new configs for the local environment/config split `lando drush cex`
5. Review config file updates `git status`
6. Confirm that the changes are made to files within the config/local directory.
7. Review the `git diff config/local/config_split.patch.core.extension.yml` file and confirm that the dblog entry has been removed.
8. Confirm that the associated dblog setting files have been removed from the config/local directory.
9. Revisit Post-deploy Step 1 and change the array key to the following: $config['config_split.config_split.non_production']
10. Run `lando drush cr && lando drush cim` to register the key change and import the config for the new environment.
11. Make changes to the site similar to steps 3 - 8 and confirm that the changes are as expected.
12. Revisit Post-deploy Step 1 and change the array key to the following: $config['config_split.config_split.production']
13. Run `lando drush cr && lando drush cim` to register the key change and import the config for the new environment.
11. Make changes to the site similar to steps 3 - 8 and confirm that the changes are as expected.

Note: If a config export is made resulting in settings that apply to all environments, then that file will be saved under config/sync directory.

## Checklist for the Developer




## Checklist for the Peer Reviewers
